File: docs/source/ampl/tutorial/ampl_files/print_solution.py
def print_solution(ampl):
    solver = ampl.get_option("solver")
    print(f"Solution with: {solver}")

    vars = ampl.get_data("_varname, _var")
    varnames = vars.get_column("_varname")
    varvals = vars.get_column("_var")
    for n, v in zip(varnames, varvals) :
        print(f"{n} = {v}")

    objs = ampl.get_data("_objname, _obj")
    objnames = objs.get_column("_objname")
    objvals = objs.get_column("_obj")
    print(f"{next(iter(objnames))} = ${next(iter(objvals)):.2f}")
    print()

File: docs/source/ampl/tutorial/ampl_files/print_solution.R
print_solution <- function(ampl) {
    solver <- ampl$getOption("solver")
    cat("Solution with:", solver, "\n")
    
    vars <- ampl$getData("_varname, _var")
    varnames <- vars$X_varname
    varvals <- vars$X_var
    mapply(function(n, v) cat(n, "=", v, "\n"), varnames, varvals)

    objs <- ampl$getData("_objname, _obj")
    objnames <- objs$X_objname
    objvals <- objs$X_obj
    cat(objnames[1], "=", sprintf("$%.2f", objvals[1]), "\n")
    cat("\n")
}
